### ISSUE TITLE
How to convert the SegmentationLoss to Euclidean Loss?

### ISSUE THREAD

**Issue by kobefeng123** [vlfeat/matconvnet#367]
I am trying to do regression using FCN, is there any easy way to convert SegmentationLoss to Euclidean Loss? As those new FCN files lack of documentations, I am quite confused on how to edit the SegmentationLoss file. Any hints would be appreciated.

**Comment by bazilas** [comment]
Hi,

If you would like to use a Euclidean Loss, you should code it by yourself and replace vl_nnloss() with your function.

V

**Comment by vedaldi** [comment]
Hi, this is coming soon. Beta18 will have it for simplenn and this will be followed by Dagnn soon after.

**Comment by cheer37** [comment]
@vedaldi 
How matconvnet can be built without gemm, gemv source?
I couldn't find that source.

**Comment by Bhee** [comment]
Euclidean Loss not found in beta 18.
can I use 'pdist ' as euclidean loss?

**Comment by vedaldi** [comment]
Hi, yes indeed. Use the ‘aggregate’, true option to that end

**Comment by Bhee** [comment]
I am trying for super resolution:  "Image Super-Resolution Using Deep Convolutional Networks
Chao Dong, Chen Change Loy, Kaiming He, Xiaoou Tang"

Architecture:
conv : weights : (3,3,1,32)  strid =1, pad =1
leaky reLU
conv : weights : (3,3,32,32)  strid =1, pad =1
leaky reLU
conv : weights : (3,3,32,64)  strid =1, pad =1
leaky reLU
conv : weights : (3,3,64,64)  strid =1, pad =1
leaky reLU
conv : weights : (3,3,64,128)  strid =1, pad =1
leaky reLU
conv : weights : (3,3,128,128)  strid =1, pad =1
leaky reLU
conv : weights : (3,3,128,1)  strid =1, pad =1
For loss fn: Euclidean loss function(pdist) with aggregate : true

Input : low resolution image ,target : high resolution image, both are 46x46 size
But  loss fn became NaN:

Training:
training: epoch 01: batch   1/700:  6.82 s (14.7 data/s) obj:0.349 [100/100]
training: epoch 01: batch   2/700:  4.75 s (21.0 data/s) obj:2.01 [100/100]
training: epoch 01: batch   3/700:  9.32 s (10.7 data/s) obj:9.94 [100/100]
training: epoch 01: batch   4/700:  15.53 s (6.4 data/s) obj:68.2 [100/100]
training: epoch 01: batch   5/700:  15.95 s (6.3 data/s) obj:1.37e+007 [100/100]
training: epoch 01: batch   6/700:  8.44 s (11.8 data/s) obj:Inf [100/100]
training: epoch 01: batch   7/700:  7.78 s (12.9 data/s) obj:NaN [100/100]

What is the reason behind it?

**Comment by vedaldi** [comment]
Hi, I would suggest to test first a much slower learning rate to see if it still diverges. Note that if the loss sums over all pixels and it is not normalised by the number of pixels, gradients can be very large.
